examples_apps/run_tutorial_machine_learning.py

```diff
@@ -0,0 +1,319 @@
+theta_bs, phi_bs = prog.params("theta_bs", "phi_bs")
+
+with prog.context as q:
+    # States
+    Coherent(alpha)            | q[0]
+    # Gates
+    BSgate(theta_bs, phi_bs)   | (q[0], q[1])
+    # Measurements
+    MeasureHomodyne(0.0)       | q[0]
+
+
+##############################################################################
+# To run a Strawberry Fields simulation with the Tensorflow backend, we need to specify
+# ``'tf'`` as the backend argument when initializing the engine:
+
+eng = sf.Engine(backend="tf", backend_options={"cutoff_dim": 7})
+
+##############################################################################
+# We can now run our program using :meth:`eng.run() <.Engine.run>`. However, directly
+# evaluating a circuit which contains symbolic parameters using :meth:`eng.run() <.Engine.run>`
+# will produce errors. The reason for this is that :meth:`eng.run() <.Engine.run>` tries,
+# by default, to numerically evaluate any measurement result. But we have not provided
+# numerical values for the symbolic arguments yet!
+#
+# .. code-block:: python3
+#
+#     eng.run(prog)
+#
+# .. rst-class:: sphx-glr-script-out
+#
+#     Out:
+#
+#     .. code-block:: none
+#
+#         ParameterError: {alpha}: unbound parameter with no default value.
+#
+# To bind a numerical value to the free parameters, we pass a mapping from parameter
+# name to value using the ``args`` keyword argument when running the engine:
+
+mapping = {
+    "alpha": tf.Variable(0.5),
+    "theta_bs": tf.constant(0.4),
+    "phi_bs": tf.constant(0.0)
+}
+
+result = eng.run(prog, args=mapping)
+
+
+##############################################################################
+# This code will execute without error, and both the output results and the
+# output samples will contain numeric values based on the given value for the angle ``phi``:
+
+print(result.samples)
+
+##############################################################################
+# We can select measurement results at other angles by supplying different values for
+# ``phi`` in ``mapping``. We can also return and perform processing on the underlying state:
+
+state = result.state
+print("Density matrix element [0,0,1,2]:", state.dm()[0, 0, 1, 2])
+
+##############################################################################
+# Computing gradients
+# -------------------
+#
+# When used within a gradient tape context, any output generated from the state
+# class can be differentiated natively using ``GradientTape().gradient()``.
+# For example, lets displace a one-mode program, and then compute the gradient
+# of the mean photon number:
+
+eng.reset()
+prog = sf.Program(1)
+
+alpha = prog.params("alpha")
+
+with prog.context as q:
+    Dgate(alpha) | q
+
+# Assign our TensorFlow variables, so that we can
+# refer to them later when differentiating/training.
+a = tf.Variable(0.43)
+
+with tf.GradientTape() as tape:
+    # Here, we pass map our quantum free parameter `alpha`

```

Comment:
```suggestion
    # Here, we map our quantum free parameter `alpha`

```

Perhaps adding `and pass it to the engine.` to the sentence instead?